src/app/signup/page.tsx

```diff
@@ -18,64 +19,90 @@ interface InvalidFields {
+    name: '',
     email: '',
     password: '',
     confirm_password: '',
     captcha: '',
     ref_code: '',
   });
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData({
+      ...formData,
+      [name]: value,
+    });
+  };
+
+  const validatePass = (rule: any, value: string, callback: any) => {
+    if (!value) {
+      callback(t('validator.password_required'));
+    } else if (formData.confirm_password) {
+      validatePassCheck(null, formData.confirm_password, callback);
+    } else {
+      callback();
+    }
+  };
+  const validatePassCheck = (rule: any, value: string, callback: any) => {
+    if (!value) {
+      callback(t('validator.confirm_password_required'));
+    } else if (value !== formData.password) {
+      callback(t('validator.password_dont_match'));
+    } else {
+      callback();
+    }
+  };
+
   const rules = () => ({
     email: [
-      { required: true, message: ('validator.account_required') },
-      { type: 'email', message: ('validator.invalid_email') },
+      { required: true, message: t('validator.account_required') },
+      { type: 'email', message: t('validator.invalid_email') },
     ],
     password: [
-      { required: true, message: ('validator.password_required') },
-      { type: 'string', min: 6, max: 32, message: ('validator.invalid_password') },
+      { required: true, message: t('validator.password_required') },
+      { validator: validatePass },
     ],
-    confirmPassword: [{ validator: validatePassCheck }],
+    confirm_password: [
+      { validator: validatePassCheck }],
     captcha: [
-      { required: true, message: ('validator.captcha_required') },
+      { required: true, message: t('validator.captcha_required') },
     ],
     ref_code: [
-      { required: false, message: ('validator.ref_required') },
+      { required: false, message: t('validator.ref_required') },
     ],
   });
   const [invalidFields, setInvalidFields] = useState<InvalidFields>({});
   const [isSubmiting, setIsSubmiting] = useState(false);
-  const formRef = useRef<any>(null);
-  const validatePass = (rule: any, value: string) => {
-    if (value === '') {
-      return Promise.reject(new Error('validator.password_required'));
-    } else {
-      if (formData.confirm_password !== '') {
-        // Verify the second password box separately
-        formRef.current.validateField(['confirmPassword']);
-      }
-      return Promise.resolve();
-    }
-  };
-  const validatePassCheck = (rule: any, value: string) => {
-    if (value === '') {
-      return Promise.reject(new Error('validator.confirm_password_required'));
-    } else if (value !== formData.password) {
-      return Promise.reject(new Error('validator.password_dont_match'
-      ));
-    } else {
-      return Promise.resolve();
-    }
+
+  const updateCaptcha = async () => {
+    const response = await axios.get('http://127.0.0.1:3000/api/v1/captcha?');
```

Comment:
Does this url work well in your test? Because it seems like you want to query a captcha in url, but nothing after the question mark